tests/functional/event_handler/test_api_gateway.py

```diff
@@ -0,0 +1,340 @@
+import base64
+import json
+import zlib
+from decimal import Decimal
+from pathlib import Path
+from typing import Dict, Tuple
+
+import pytest
+
+from aws_lambda_powertools.event_handler.api_gateway import ApiGatewayResolver, ProxyEventType, Response
+from aws_lambda_powertools.shared.json_encoder import Encoder
+from aws_lambda_powertools.utilities.data_classes import ALBEvent, APIGatewayProxyEvent, APIGatewayProxyEventV2
+from tests.functional.utils import load_event
+
+
+def read_media(file_name: str) -> bytes:
+    path = Path(str(Path(__file__).parent.parent.parent.parent) + "/docs/media/" + file_name)
+    return path.read_bytes()
+
+
+LOAD_GW_EVENT = load_event("apiGatewayProxyEvent.json")
+TEXT_HTML = "text/html"
+APPLICATION_JSON = "application/json"
+
+
+def test_alb_event():
```

Comment:
could you split that into a separate file? For example, [ALB has a maximum of 1MB response size](https://docs.aws.amazon.com/elasticloadbalancing/latest/application/lambda-functions.html#prepare-lambda-function) -- Do we want to take limits like that into account as ALB and API GW differ?.

ALB also supports [healthchecks](https://docs.aws.amazon.com/elasticloadbalancing/latest/application/lambda-functions.html#enable-health-checks-lambda) (disabled by default), but I think that's better to do in Dataclass so one can easily check for that in a method  by inspecting the user header (don't think is necessary but throwing it out, etc.

Reply:
@heitorlessa do you know what’s the limit for the response size in the API GW? When you setup the resolver  you do have to specify the type of integration. So we could to handle it there? Or do you want a class per integration type? It seems unneeded.

Reply:
API GW would be 10MB but effectively Lambda limits synchronous requests to 6MB. So 6M for API GW and 1M for Lambda.

I have second thoughts on whether we should account for these, lemme ask @pcolazurdo or @am29d on this -- Do you think we should raise an Exception if the Lambda response is bigger than the event source supports OR let it explicitly fail as it is today?

Since we're not technically a framework and don't handle infra but app code, I'm heading towards the latter - Let API GW or ALB handle that as we don't have control over these limits and will unlikely be in the known on a timely fashion if these ever get up to date.

Reply:
I would let it fail as it is today, the error message is clear and it is easy to spot the source and fix it.

Reply:
So I am leaving this as is for now